src/api/spec/models/staging/workflow_spec.rb

```diff
@@ -83,21 +80,27 @@
       it { expect(subject).to be_empty }
     end
 
-    context 'with requests and some are in staging projects and some not' do
-      let!(:bs_request_2) do
+    context 'with request in state review whether they are in staging projects or not' do
+      let(:bs_request_2) do
         create(:bs_request_with_submit_action,
+               state: :review,
                target_project: project.name,
                target_package: target_package.name,
                source_project: source_project.name,
                source_package: source_package.name)
       end
 
       before do
-        bs_request.staging_project = staging_project
-        bs_request.save
+        bs_request
+        bs_request_2.change_review_state(:accepted, by_group: group.title)
+        # A review by project is created by default. See BsRequest#apply_default_reviewers. We have to accept it too.
+        bs_request_2.change_review_state(:accepted, by_project: source_project.name)
+        bs_request_2.staging_project = staging_project
+        bs_request_2.save
       end
 
       it { expect(subject).to contain_exactly(bs_request_2) }
+      it { expect(subject.first.state).to eq(:new) }
```

Comment:
as you already verified that it's only bs_request_2, you're testing change_review_state here and not the staging controller. As such: I see no value of the 2nd expect